docs/src/main/tut/dsl.md

```diff
@@ -392,4 +392,48 @@ val averageTemperatureService = HttpService[IO] {
 }
 ```
 
+#### Optional query parameters
+
+To accept a optional query parameter a `OptionalQueryParamDecoderMatcher` can be used.
+
+```tut:book
+object OptionalYearQueryParamMatcher extends OptionalQueryParamDecoderMatcher[Year]("name")
+
+def getAverageTemperatureForCurrentYear(): IO[String] = ???
```

Comment:
Nit: if it returns a pure value, Scala style is to omit the parens.

Reply:
Its probably not pure. The evaluation of "CurrentYear" has a side effect =P